File: 0016-3sum-closest/0016-3sum-closest.java
class Solution {
    public int threeSumClosest(int[] nums, int target) {
          int mini = nums[0] + nums[1] + nums[2];
        Arrays.sort(nums);
        int l=0,r=0,n=nums.length;
        for(int i=0;i<n-2;i++){
            l=i+1;r=n-1;
         //   mini=Math.min(Math.abs(sum-target),mini);
            while(l<r){
                 int sum=nums[i]+nums[l]+nums[r];
                 if (Math.abs(sum - target) < Math.abs(mini - target)) {
                    mini = sum;
                }
                if(sum==target)return target;
                else if(sum<target){
                    l++;
                }
                else if(sum>target){
                    r--;
                }
            }
        }
        return mini;
    }
}

File: 0016-3sum-closest/0016-3sum-closest.py
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        nums.sort()
        n=len(nums)
        diff=nums[n-2]+nums[n-1]+nums[n-3]

        for i in range(n-2):
            l=i+1
            r=n-1

            while(l<r):
                curr_sum=nums[i]+nums[l]+nums[r]
                if curr_sum>target:
                    r-=1
                elif curr_sum<target:
                    l+=1
                else:
                    return target
                if abs(curr_sum - target) < abs(diff - target):diff = curr_sum
        return diff